### PR TITLE
Added preliminary support for NextSeq

### PR DESCRIPTION
**Pull request by jfnavarro**
Cleaned up the code a bit
Formated the code a bit
Made the transfer to Uppmax optional

### CODE REVIEW

**Comment by Galithil** [comment]
I'm not this package maintainer, but here are my thoughts : 

removing import csv might be a mistake

removing empty lines in docstrings usually mess up sphinx output

typo in analysis.py line 91

If you with to reformat multi line comments, please use three double quotes

I don't really agree with the removal of empty lines between imports of different packages

I'm also not sure about some of the new logic involved in analysis.py:190

I'll continue later

File: taca/analysis/analysis.py [hunk 1]
```diff
@@ -1,246 +1,218 @@
...
-            return 'HiSeq'
+            return MiSeq_Run(run, CONFIG["analysis"]["MiSeq"])
+        elif "NextSeq" in runtype:
+            return NextSeq_Run(run, CONFIG["analysis"]["NextSeq"])
         else:
-            logger.warn("unrecognized runtype {}, cannot archive the run {}. Someone as likely bought a new sequencer without telling it to the bioinfo team".format(runtype, run))
-            return None
+            logger.warn("Unrecognized run type {}, cannot archive the run {}. " + 
+                        "Someone as likely bought a new sequencer without telling " + 
+                        "it to the bioinfo team".format(runtype, run))
```
Comment (Galithil) [inline taca/analysis/analysis.py]: I'm fine with people following pep8, but not with breaking up a string  and using concatenation. Either use the backslash or the implicit concatenation.

**Comment by Galithil** [comment]
A lot of the changes are rather odd.
Some `return None` where replaced with just `return`, some others were simply removed. Some lines were wrapped to fit 80 cols, some lines were unwrapped. 

Some of the arguments were right-aligned, and now they aren't anymore.

I disagree with changing the whole coding style of the repo over one commit, especially one that happens to change 14 files and break the build.

**Comment by jfnavarro** [comment]
Oh thanks for the comments :) 
I am just trying to set the development environment this is far to be completed
or something that should be considered to be merged into the master, I have not even
tested it as it is just a development branch. Maybe I should have not
made a pull request to Scilifelab and keep working on my repo, 
I am not very famliar with github. I just want to have a branch where I can work
and develop and when it is tested and cleaned merge it to the master if you guys agree.

Regarding code style, formatting and typos. I see different code styles
and formatting in the code. Very long lines, typos, etc... I tend to fix
those things just automatically. 

Can you send me the coding and style guidelines? Do you use the Python
official coding and formatting guidelines or just personal preferences here and there?

**Comment by Galithil** [comment]
Guillermo used to recommend reading this before doing anything else : https://github.com/guillermo-carrasco/BestPracticesWorkshop

The main guidelines in Python are the PEP https://www.python.org/dev/peps/

If this is far from completed, please refrain from making Pull Requests to the master branch. The master branch is the code we deploy and run in production, and therefore, any change to it is closely monitored. 
I strongy recommend that you activate Travis testing if you work on taca, as any Pull Request that fail travis tests will likely not be merged.

When you make a pull request to master, the message we get is "This code is production ready and I want it deployed". If that is not the message you wish to send, consider making a pull request to a development branch.

Most of the time, we do most of the development work on our personal forks, as it limits the amount of branches the main repo has, and the code is still very easy to share. 

Regarding code style and formatting : the main issue here is that this repo has had 8 different developers. If every single one of them "corrects" the style of a part of another developer every time they work on something, it makes it impossible to follow the changes, or even review the code.

Again, I am not this package's maintainer, @vezzi is the one you need to have a chat with regarding how to contribute to this repo.

**Pull request closed without merging**